src/utilities.lisp

```diff
@@ -0,0 +1,24 @@
+;;;; utilities.lisp
+;;;;
+;;;; Author: Eric Peterson
+
+(in-package #:rpcq)
+
+
+(defun sanitize-name (symb)
+  "Convert a stringifiable SYMB into a snake-case string."
+  (substitute #\_ #\- (string symb)))
+
+(defun str->lisp-keyword (str)
+  "Convert a snake-case string into a hyphenated Lisp keyword."
+  (make-keyword (substitute #\- #\_ (string-upcase str))))
+
+(defun format-log (s fmt-string &rest args)
+  "Writes a format string to the stream S in debug output format."
+  (format s "[~A | ~A] ~?" (local-time:now) (bt:thread-name (bt:current-thread)) fmt-string args)
+  (terpri s))
+
+(defun unpack-foreign-msg-to-bytes (msg)
+  "Converts a foreign array of unsigned characters to a Lisp vector of such."
+  (cffi:foreign-array-to-lisp (pzmq:msg-data msg)
+                              `(:array :unsigned-char ,(pzmq:msg-size msg))))
```

Comment:
make sure decision about `unsigned-char` (vs uint8) is consistent here

Reply:
I think this https://github.com/cffi/cffi/blob/master/src/types.lisp#L1048 means they're the same?

Reply:
should not be relied upon

Reply:
well ok